tests/src/integration/setup.rs

```diff
@@ -0,0 +1,167 @@
+use std::mem::ManuallyDrop;
+use std::path::Path;
+use std::str::FromStr;
+use std::sync::{Arc, Mutex};
+
+use color_eyre::eyre::{eyre, Result};
+use namada_apps::cli::args;
+use namada_apps::config;
+use namada_apps::config::genesis::genesis_config;
+use namada_apps::config::genesis::genesis_config::GenesisConfig;
+use namada_apps::config::TendermintMode;
+use namada_apps::facade::tendermint::Timeout;
+use namada_apps::facade::tendermint_proto::google::protobuf::Timestamp;
+use namada_apps::node::ledger::shell::Shell;
+use namada_core::types::address::Address;
+use namada_core::types::chain::{ChainId, ChainIdPrefix};
+use toml::value::Table;
+
+use super::node::MockNode;
+use crate::e2e::setup::{
+    copy_wasm_to_chain_dir, get_all_wasms_hashes, TestDir, ENV_VAR_KEEP_TEMP,
+    SINGLE_NODE_NET_GENESIS,
+};
+
+/// Setup a network with a single genesis validator node.
+pub fn setup() -> Result<MockNode> {
+    initialize_genesis(|genesis| genesis)
+}
+
+/// Setup folders with genesis, configs, wasm, etc.
+pub fn initialize_genesis(
+    mut update_genesis: impl FnMut(GenesisConfig) -> GenesisConfig,
+) -> Result<MockNode> {
+    let working_dir = std::fs::canonicalize("..").unwrap();
+    let keep_temp = match std::env::var(ENV_VAR_KEEP_TEMP) {
+        Ok(val) => val.to_ascii_lowercase() != "false",
+        _ => false,
+    };
+    let test_dir = {
+        std::env::set_var(ENV_VAR_KEEP_TEMP, "true");
```

Comment:
this re-use of `TestDir` from e2e seems to forcibly add unnecessarily coupling and unsafe code - maybe we do it better without it?